internal/processmon/processmon.go

```diff
@@ -0,0 +1,379 @@
+		zap.L().Warn("Failed to remote process from cache", zap.Error(err))
+	}
+}
+
+// pollStdOutAndErr polls std out and err
+func (p *processMon) pollStdOutAndErr(
+	cmd *exec.Cmd,
+	exited chan int,
+	contextID string,
+) (initializedCount int, err error) {
+
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return initializedCount, err
+	}
+
+	initializedCount++
+
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return initializedCount, err
+	}
+
+	initializedCount++
+
+	// Stdout/err processing
+	go processIOReader(stdout, contextID, exited)
+	go processIOReader(stderr, contextID, exited)
+
+	return initializedCount, nil
+}
+
+// getLaunchProcessCmd returns the command used to launch the enforcerd
+func (p *processMon) getLaunchProcessCmd(arg string, contextID string) (*exec.Cmd, error) {
+
+	cmdName, err := osext.Executable()
+	if err != nil {
+		return nil, err
+	}
+
+	cmdArgs := []string{arg}
+	cmdArgs = append(cmdArgs, p.launcProcessArgs...)
+	if !p.logToConsole {
+		cmdArgs = append(cmdArgs, contextID)
+	}
+	zap.L().Debug("Enforcer executed",
+		zap.String("command", cmdName),
+		zap.Strings("args", cmdArgs),
+	)
+
+	return exec.Command(cmdName, cmdArgs...), nil
+}
+
+// getLaunchProcessEnvVars returns a slice of env variable strings where each string is in the form of key=value
+func (p *processMon) getLaunchProcessEnvVars(
+	procMountPoint string,
+	contextID string,
+	randomkeystring string,
+	statsServerSecret string,
+	refPid int,
+	refNSPath string,
+) []string {
+
+	mountPoint := constants.AporetoEnvMountPoint + "=" + procMountPoint
+	namedPipe := constants.AporetoEnvContextSocket + "=" + contextID2SocketPath(contextID)
+	statsChannel := constants.AporetoEnvStatsChannel + "=" + rpcwrapper.StatsChannel
+	rpcClientSecret := constants.AporetoEnvRPCClientSecret + "=" + randomkeystring
+	envStatsSecret := constants.AporetoEnvStatsSecret + "=" + statsServerSecret
+	containerPID := constants.AporetoEnvContainerPID + "=" + strconv.Itoa(refPid)
+
+	newEnvVars := []string{
+		mountPoint,
+		namedPipe,
+		statsChannel,
+		rpcClientSecret,
+		envStatsSecret,
+		containerPID,
+	}
+
+	// If the PURuntime Specified a NSPath, then it is added as a new env var also.
+	if refNSPath != "" {
+		nsPath := constants.AporetoEnvNSPath + "=" + refNSPath
+		newEnvVars = append(newEnvVars, nsPath)
+	}
+
+	return newEnvVars
+}
+
+// LaunchProcess prepares the environment and launches the process
+func (p *processMon) LaunchProcess(
+	contextID string,
+	refPid int,
+	refNSPath string,
+	rpchdl rpcwrapper.RPCClient,
+	arg string,
+	statsServerSecret string,
+	procMountPoint string,
+) error {
+
+	_, err := p.activeProcesses.Get(contextID)
+	if err == nil {
+		return nil
+	}
+
+	var nsPath string
+
+	// We check if the NetNsPath was given as parameter.
+	// If it was we will use it. Otherwise we will determine it based on the PID.
+	if refNSPath == "" {
+		nsPath = filepath.Join(procMountPoint, strconv.Itoa(refPid), "ns/net")
+	} else {
```

Comment:
you can skip the else by doing 

```go
nsPath := refNSPath
if nsPath == "" {
    nsPath = filepath.Join(procMountPoint, strconv.Itoa(refPid), "ns/net")
}

Reply:
dont agree. This is 2 assignments and 1 compare. The other is one 1 compare and assign always.